cf-custom-resources/lib/nlb-cert-validator-updater.js

```diff
@@ -0,0 +1,333 @@
+    // managed by the custom resource lambda; on the contrary, the same set of aliases indicate that there is no need to
+    // replace or update the certificate, nor the validation records or A records. Hence, we can use this as the physicalResourceID.
+    let aliasesSorted = [...aliases].sort().join(",");
+    const physicalResourceID = `/${serviceName}/${aliasesSorted}`;
+
+    let handler = async function() {
+        switch (event.RequestType) {
+            case "Create":
+                await validateAliases(aliases, loadBalancerDNS);
+                const certificateARN = await requestCertificate({
+                    aliases: aliases,
+                    idempotencyToken: CRYPTO
+                        .createHash("sha256")
+                        .update(physicalResourceID)
+                        .digest("hex")
+                        .substr(0, 32),});
+                const options = await waitForValidationOptionsToBeReady(certificateARN, aliases);
+                await activate(options, certificateARN, loadBalancerDNS, loadBalancerHostedZoneID);
+                break;
+            case "Update":
+            case "Delete":
+            default:
+                throw new Error(`Unsupported request type ${event.RequestType}`);
+        }
+    };
+
+    try {
+        await Promise.race([exports.deadlineExpired(), handler(),]);
+        await report(event, context, "SUCCESS", physicalResourceID);
+    } catch (err) {
+        console.log(`Caught error for service ${serviceName}: ${err.message}`);
+        await report(event, context, "FAILED", physicalResourceID, null, err.message);
+    }
+};
+
+/**
+ * Validate that the aliases are not in use.
+ *
+ * @param {Set<String>} aliases for the service.
+ * @param {String} loadBalancerDNS the DNS of the service's load balancer.
+ * @throws error if at least one of the aliases is not valid.
+ */
+async function validateAliases(aliases, loadBalancerDNS) {
+    let promises = [];
+
+    for (let alias of aliases) {
+        const promise = envRoute53.listResourceRecordSets({
+            HostedZoneId: envHostedZoneID,
+            MaxItems: "1",
+            StartRecordName: alias,
+        }).promise().then((data) => {
+            let recordSet = data["ResourceRecordSets"];
+            if (!recordSet || recordSet.length === 0) {
+                return;
+            }
+            let aliasTarget = recordSet[0].AliasTarget;
+            if (aliasTarget && aliasTarget.DNSName === `${loadBalancerDNS}.`) {
+                return; // The record is an alias record and is in use by myself, hence valid.
+            }
+
+            if (aliasTarget) {
+                throw new Error(`Alias ${alias} is in use by ${aliasTarget.DNSName}. This could be another load balancer of a different service.`);
+            }
+            throw new Error(`Alias ${alias} is in use`);
```

Comment:
nit:
```suggestion
                throw new Error(`Alias ${alias} is already in use by ${aliasTarget.DNSName}. This could be another load balancer of a different service.`);
            }
            throw new Error(`Alias ${alias} is already in use`);
```